### PR TITLE
PLAT-5389: Fix for fast disconnections and support to _publish entries

### PR DESCRIPTION
**Pull request by gadyaari**
…treams

### CODE REVIEW

File: lib/Worker.js [hunk 1]
```diff
@@ -80,23 +80,23 @@ module.exports = (function(){
...
+                            logger.error("[%s] Error adding entry to endlist appender processing - %s", e.entryId, err.message);
                         });
                 }
                 else {
-                    logger.info("Entry creation aborted for entry %s", e.entryId);
+                    logger.info("[%s] Entry creation aborted", e.entryId);
                     entry.stop()
                         .catch(function(err) {
-                            logger.error("error aborting liveEntry object creation - %s", err);
+                            logger.error("[%s] Aborting liveEntry object creation - %s", e.entryId, err);
```
Comment (kalturaguy) [inline lib/Worker.js]: use %j

File: lib/Worker.js [hunk 2]
```diff
@@ -110,6 +110,10 @@ module.exports = (function(){
         });
         _.forEach(alreadyLiveEntries, function(e) {
             var entry = that.handledEntries[e.entryId];
+            if (entry === 'cancelPending') {
+                entry = 'pending';
```
Comment (kalturaguy) [inline lib/Worker.js]: that.handledEntries[e.entryId]='pending'

File: lib/Worker.js [hunk 3]
```diff
@@ -208,7 +218,7 @@ module.exports = (function(){
                 var relevantEntries = _.filter(liveEntries, function (e) {
                     return e.entryId.indexOf(that.prefix) === 0;
                 });
-                logger.debug('relevant live entries: ' + JSON.stringify(relevantEntries));
+                logger.debug("Entries received from Wowza: %j", JSON.stringify(relevantEntries));
```
Comment (kalturaguy) [inline lib/Worker.js]: no need for JSON.strinfy

File: lib/Worker.js (hunk 1 above)
Comment (gadyaari) [inline lib/Worker.js]: done

File: lib/Worker.js (hunk 2 above)
Comment (gadyaari) [inline lib/Worker.js]: done

File: lib/Worker.js (hunk 3 above)
Comment (gadyaari) [inline lib/Worker.js]: done

File: lib/Worker.js [hunk 4]
```diff
@@ -110,47 +105,52 @@ module.exports = (function(){
...
-                        // If the liveEntry creation was unsuccessful, allow it to be created again
-                        // in the next round
-                        var liveEntryCreationTryCount = that.phantomEntries[e.entryId] || 0;
-                        if (!that.phantomEntries[e.entryId]) {
-                            logger.info("Adding entry %s to the phantom entries list", e.entryId);
+                        logger.error("[%s] Error starting entry object: %s, %s", e.entryId, err.message, err.stack);
+                        if (that.handledEntries[e.entryId] === 'cancelPending') {
+                            // Entry stopped streaming while trying to create object for it. Ignore the error and delete it from list
+                            logger.info("[%s] Ignoring error since entry stopped streaming", e.entryId);
                         }
```
Comment (kalturaguy) [inline lib/Worker.js]: shouldn't we return here?

**Pull request merged**